tensorflow/core/kernels/non_max_suppression_op_test.cc

```diff
@@ -355,6 +355,178 @@ TEST_F(NonMaxSuppressionV2OpTest, TestEmptyInput) {
   test::ExpectTensorEqual<int>(expected, *GetOutput(0));
 }
 
+#if GOOGLE_CUDA
+class NonMaxSuppressionV2GPUOpTest : public OpsTestBase {
+ protected:
+  void MakeOp() {
+    SetDevice(DEVICE_GPU,
+              std::unique_ptr<tensorflow::Device>(DeviceFactory::NewDevice(
+                  "GPU", {}, "/job:a/replica:0/task:0")));
+
+    TF_EXPECT_OK(
+        NodeDefBuilder("non_max_suppression_op_gpu", "NonMaxSuppressionV2")
+            .Input(FakeInput(DT_FLOAT))
+            .Input(FakeInput(DT_FLOAT))
+            .Input(FakeInput(DT_INT32))
+            .Input(FakeInput(DT_FLOAT))
+            .Finalize(node_def()));
+    TF_EXPECT_OK(InitOp());
+  }
+};
+
+TEST_F(NonMaxSuppressionV2GPUOpTest, TestSelectFromThreeClusters) {
+  MakeOp();
+  AddInputFromArray<float>(
+      TensorShape({6, 4}),
+      {0, 0,  1, 1,  0, 0.1f,  1, 1.1f,  0, -0.1f, 1, 0.9f,
```

Comment:
Add a comment saying that these are box coordinates and per-line comments for the corresponding scores (like below), same below.